### PR TITLE
chore: update repo

### PR DESCRIPTION
**Pull request by pi0**
Summary:

- Simplify repo and remove unnecessary stuff
- Use rollup build to tree-shake only used logic into single file (also exposing public types)
- Add explicit dependencies and peer dependencies
- Temporary disable module e2e test until fixing typescript-build and nuxt dependency with proper meta
- Update typescript to 3.9

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ export interface Options {
 const isPureObjectOrString = (val: any) => (!Array.isArray(val) && typeof val === 'object') || typeof val === 'string'
 const getDir = (p: string) => fs.statSync(p).isDirectory() ? p : path.dirname(p)
 
-export default <Module> function () {
+const compnentsModule = <Module> function () {
```
Comment (kevinmarrec) [inline src/index.ts:38]:
```suggestion
const componentsModule = <Module> function () {
```

File: src/index.ts [hunk 2]
```diff
@@ -133,7 +133,12 @@ export default <Module> function () {
...
   this.nuxt.hook('webpack:config', (configs: WebpackConfig[]) => {
     for (const config of configs.filter(c => ['client', 'modern', 'server'].includes(c.name!))) {
-      ((config.entry as WebpackEntry).app as string[]).unshift(path.resolve(__dirname, 'installComponents.js'))
+      ((config.entry as WebpackEntry).app as string[]).unshift(path.resolve(__dirname, '../lib/installComponents.js'))
     }
   })
 }
+
+// @ts-ignore
+compnentsModule.meta = { name: '@nuxt/components' }
```
Comment (kevinmarrec) [inline src/index.ts:142]:
```suggestion
componentsModule.meta = { name: '@nuxt/components' }
```

File: src/index.ts [hunk 3]
```diff
@@ -133,7 +133,12 @@ export default <Module> function () {
...
+
+export default compnentsModule
```
Comment (kevinmarrec) [inline src/index.ts:144]:
```suggestion
export default componentsModule
```

**Comment by pi0** [comment]
Sorry for breaking 100 coverage :broken_heart:

**Pull request merged**